src/main/scala/ch/epfl/bluebrain/nexus/kg/storage/package.scala

```diff
@@ -6,24 +6,28 @@ import java.security.MessageDigest
 import java.util.UUID
 
 import akka.http.scaladsl.model.Uri
-import akka.stream.scaladsl.{Sink, Source}
+import akka.stream.scaladsl.{Flow, Keep, Sink, Source}
 import akka.util.ByteString
 import ch.epfl.bluebrain.nexus.kg.resources.ProjectRef
+import ch.epfl.bluebrain.nexus.kg.resources.file.File.Digest
 
 import scala.concurrent.Future
 
 package object storage {
   type AkkaSource = Source[ByteString, Any]
 
   /**
-    * A sink that computes the digest of the ByteString
+    * A sink that computes the digest of the input ByteString
+    *
     * @param algorithm the digest algorithm. E.g.: SHA-256
     */
-  def digestSink(algorithm: String): Sink[ByteString, Future[MessageDigest]] =
-    Sink.fold(MessageDigest.getInstance(algorithm))((digest, currentBytes) => {
-      digest.update(currentBytes.asByteBuffer)
-      digest
-    })
+  def digestSink(algorithm: String): Sink[ByteString, Future[Digest]] = {
+    val digest = MessageDigest.getInstance(algorithm)
+    Flow[ByteString]
+      .map(bs => digest.update(bs.asByteBuffer))
+      .map(_ => Digest(digest.getAlgorithm, digest.digest.map("%02x".format(_)).mkString))
```

Comment:
Doesn't this perform the `digest.digest.map("%02x".format(_)` for every chunk on the Source ?
Also, `Sink.head`
> Materializes into a Future which completes with the first value arriving, after this the stream is canceled

which doesn't seem to be what we want to do here.

Why not:
```scala
def digestSink(algorithm: String): Sink[ByteString, Future[Digest]] =
      Sink
        .fold(MessageDigest.getInstance(algorithm))((digest, currentBytes: ByteString) => {
          digest.update(currentBytes.asByteBuffer)
          digest
        })
        .mapMaterializedValue(_.map(dig => Digest(dig.getAlgorithm, dig.digest().map("%02x".format(_)).mkString)))
```